packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata-validation.service.ts

```diff
@@ -31,18 +33,22 @@ export class FieldMetadataValidationService<
   }
 
   private validateNumberSettings(settings: FieldMetadataSettings<T>) {
-    if ('decimals' in settings) {
-      const { decimals } = settings;
-
-      if (
-        decimals !== undefined &&
-        (decimals < 0 || !Number.isInteger(decimals))
-      ) {
-        throw new FieldMetadataException(
-          `Decimals value "${decimals}" must be a positive integer`,
-          FieldMetadataExceptionCode.INVALID_FIELD_INPUT,
-        );
-      }
+    const schema = z.object({
```

Comment:
nit: I prefer to avoid zod in the backend and use class-validator instead since backend is more class oriented. It seems the package is already there so it's fine but I'll add to my TODO so we don't have both libs of validation and class-validator makes more sense in general so we will keep that one!

Reply:
Try with this if you want
```typescript
 enum ValueType {
  PERCENTAGE = 'percentage',
  NUMBER = 'number'
}

class SettingsValidation {
  @IsOptional()
  @IsInt()
  @Min(0)
  decimals?: number;

  @IsOptional()
  @IsEnum(ValueType)
  type?: 'percentage' | 'number';
}

async function validateSettings(settings: any) {
  try {
    const settingsInstance = plainToClass(SettingsValidation, settings);
    
    await validateOrReject(settingsInstance);
  } catch (errors) {
    const errorMessages = errors
      .map((error: any) => Object.values(error.constraints))
      .flat()
      .join(', ');
      
    throw new FieldMetadataException(
      `Value for settings is invalid: ${errorMessages}`,
      FieldMetadataExceptionCode.INVALID_FIELD_INPUT
    );
  }
}
```

Reply:
+1